search/tasks.py

```diff
@@ -689,6 +841,237 @@ def start_recreate_index(self, indexes):
     )
 
 
+@app.task(bind=True)
+def start_update_index(self, indexes, platform):
```

Comment:
This task is fairly large, I think it would be cleaner to instead fan out into a set of per-object-type tasks and then have each of those fan out further using the per-type code below.